core/devmode-spi/src/main/java/io/quarkus/dev/testing/TestWatchedFiles.java

```diff
@@ -11,19 +14,23 @@
 public class TestWatchedFiles {
 
     private static volatile Map<String, Boolean> watchedFilePaths;
-    private static volatile Consumer<Map<String, Boolean>> watchedFilesListener;
+    private static volatile BiConsumer<Map<String, Boolean>, List<Entry<Predicate<String>, Boolean>>> watchedFilesListener;
+    private static volatile List<Entry<Predicate<String>, Boolean>> watchedFilePredicates;
 
-    public synchronized static void setWatchedFilePaths(Map<String, Boolean> watchedFilePaths) {
+    public synchronized static void setWatchedFilePaths(Map<String, Boolean> watchedFilePaths,
```

Comment:
I'm not so sure about these changes. It's a breaking change in the `devmode-spi`. And I have no idea if there's something out there using this SPI.

Reply:
```java
@Deprecated(forRemoval = true)
public synchronized static void setWatchedFilePaths(Map<String, Boolean> watchedFilePaths) {
    setWatchedFilePaths(watchedFilePaths, List.of());
}
```

should preserve compatibility pretty easily.

Reply:
Of course, but I'm not sure it's worth the effort. I don't even grasp the meaning of `TestWatchedFiles`  :shrug:

Reply:
But you're probaly right. Let's deprecate the old methods...